project_outsourcing/task_outsourcing_tab.py

```diff
@@ -0,0 +1,43 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class ProjectTask(models.Model):
+    """Add the fields contained in the outsourcing tab on the task."""
+
+    _inherit = 'project.task'
+
+    outsourcing_po_ids = fields.One2many(
+        'purchase.order',
+        'task_id',
+        'Outsourcing Purchase Orders',
+    )
+
+    outsourcing_line_ids = fields.Many2many(
+        'purchase.order.line',
+        compute='_compute_outsourcing_line_ids',
+        string='Outsourcing Purchase Lines',
+    )
+
+    def _compute_outsourcing_line_ids(self):
+        for task in self:
+            task.outsourcing_line_ids = task.mapped('outsourcing_po_ids.order_line')
+
+
+class PurchaseOrder(models.Model):
+
+    _inherit = 'purchase.order'
+
+    @api.onchange('task_id')
+    def _onchange_task_set_project(self):
+        """Set the project if different from the task.
```

Comment:
Codacy found an issue: [Multi-line docstring summary should start at the second line (D213)](https://app.codacy.com/gh/Numigi/odoo-project-addons/pullRequest?prid=6119862)